src/Stryker.Core/Stryker.Core/Initialisation/ProjectComponentsBuilder.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using Stryker.Core.Exceptions;
+using Stryker.Core.InjectedHelpers;
+using System.Xml.Linq;
+using System.Linq;
+using Stryker.Core.ProjectComponents;
+using Buildalyzer;
+using System.IO;
+using System.IO.Abstractions;
+
+namespace Stryker.Core.Initialisation
+{
+    public abstract class ProjectComponentsBuilder
+    {
+        public abstract IProjectComponent Build();
+
+        protected IEnumerable<string> ExtractProjectFolders(IAnalyzerResult projectAnalyzerResult, IFileSystem _fileSystem)
```

Comment:
The IFileSystem instance should be provided using a constructor on this class and calling super from the extending builders.